mrp/src/mrp/runtime/conda.py

```diff
@@ -278,64 +284,72 @@ def _env_name(self):
             """The target environment name."""
             return self.use_named_env or f"mrp_{self.name}"
 
-        def _cache_valid(self, yaml_path, env_content):
-            """Check if the cache is valid.
+        def _config_path(self):
+            dirpath = os.path.expanduser(f"~/.config/mrp/conda/{self._env_name()}/")
+            os.makedirs(dirpath, exist_ok=True)
+            return dirpath
 
-            To be valid, the cache must:
-            - have the named yaml file.
-            - have the same channels and dependencies as generated in the last up command.
-            - have a single command in the conda env history.
-            - the command must match the creation command.
-            """
-            try:
-                if not os.path.exists(yaml_path):
-                    return False
+        def _yaml_path(self):
+            return os.path.join(self._config_path(), "conda_env.yaml")
 
-                old_content = json.load(open(yaml_path))
+        def _setup_commands_snapshot_path(self):
+            return os.path.join(self._config_path(), "setup_commands.snapshot")
 
-                old_content_compare_key = json.dumps(old_content, sort_keys=True)
-                new_content_compare_key = json.dumps(env_content, sort_keys=True)
-                if old_content_compare_key != new_content_compare_key:
-                    return False
+        def _conda_history_snapshot_path(self):
+            return os.path.join(self._config_path(), "history.snapshot")
 
-                info = json.loads(
-                    subprocess.check_output(
-                        ["conda", "env", "export", "--json", "-n", self._env_name()]
-                    )
+        def _conda_history_path(self):
+            info = json.loads(
+                subprocess.check_output(
+                    ["conda", "env", "export", "--json", "-n", self._env_name()]
                 )
+            )
 
-                history_file = os.path.join(info["prefix"], "conda-meta/history")
-                if not os.path.exists(history_file):
+            return os.path.join(info["prefix"], "conda-meta/history")
```

Comment:
Maybe use simpler conda commands?
```
(base) tingfan-mbp:Downloads tingfan$ conda info --json --base
{
  "root_prefix": "/Users/tingfan/miniconda3"
}
(base) tingfan-mbp:Downloads tingfan$ 
```

Reply:
This finds the common root, but not the env specific prefix.
The env specific prefix is usually, but not always `{root_prefix}/{env_name}`